modules/ort/src/main/java/org/eclipse/sw360/antenna/ort/resolver/OrtResultArtifactResolver.java

```diff
@@ -0,0 +1,154 @@
+import org.eclipse.sw360.antenna.model.artifact.facts.javaScript.JavaScriptCoordinates;
+import org.eclipse.sw360.antenna.model.xml.generated.MatchState;
+import org.eclipse.sw360.antenna.util.LicenseSupport;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class OrtResultArtifactResolver implements Function<Package, Artifact> {
+    private SortedMap<Identifier, Map<LicenseFinding, List<PathExclude>>> licenseFindings;
+
+    public OrtResultArtifactResolver(OrtResult result) {
+        licenseFindings = result.collectLicenseFindings(false);
+    }
+
+    @Override
+    public Artifact apply(Package pkg) {
+        Artifact a = new Artifact("OrtResult")
+                .addFact(new ArtifactMatchingMetadata(MatchState.EXACT));
+
+        mapCoordinates(pkg).ifPresent(a::addFact);
+        mapSourceUrl(pkg).ifPresent(a::addFact);
+
+        mapDeclaredLicense(pkg).ifPresent(a::addFact);
+        mapFilename(pkg).ifPresent(a::addFact);
+        mapHomepage(pkg).ifPresent(a::addFact);
+
+        mapObservedLicense(pkg).ifPresent(a::addFact);
+        mapCopyrights(pkg).ifPresent(a::addFact);
+
+        return a;
+    }
+
+    private static Optional<ArtifactCoordinates> mapCoordinates(Package pkg) {
+        String namespace = pkg.getId().getNamespace();
+        String name = pkg.getId().getName();
+        String version = pkg.getId().getVersion();
+
+        switch (pkg.getId().getType().toLowerCase()) {
+            case "nuget":
+            case "dotnet":
+                return Optional.of(mapDotNetCoordinates(name, version));
+            case "maven":
+                return Optional.of(mapMavenCoordinates(namespace, name, version));
+            case "npm":
+                return Optional.of(mapJavaScriptCoordinates(name, version));
+            default:
+                return Optional.of(mapSimpleCoordinates(name, version));
+        }
+    }
+
+    private static ArtifactCoordinates mapDotNetCoordinates(String name, String version) {
+        DotNetCoordinates.DotNetCoordinatesBuilder c = new DotNetCoordinates.DotNetCoordinatesBuilder();
+        c.setPackageId(name);
+        c.setVersion(version);
+        return c.build();
+    }
+
+    private static ArtifactCoordinates mapMavenCoordinates(String namespace, String name, String version) {
+        MavenCoordinates.MavenCoordinatesBuilder c = new MavenCoordinates.MavenCoordinatesBuilder();
+        c.setGroupId(namespace);
+        c.setVersion(version);
+        c.setArtifactId(name);
+        return c.build();
+    }
+
+    private static ArtifactCoordinates mapJavaScriptCoordinates(String name, String version) {
+        JavaScriptCoordinates.JavaScriptCoordinatesBuilder c = new JavaScriptCoordinates.JavaScriptCoordinatesBuilder();
+        c.setName(name);
+        c.setVersion(version);
+        c.setArtifactId(name + "-" + version);
```

Comment:
Is that default behaviour? Maybe it would be good to implement that in the `JavaScriptCoordinates` as a fallback if `artifactId` is unset.

Reply:
This code was just moved from how / where it was before my commit, so I'd like to keep this as-is for now.